### DISCUSSION TITLE
Identify all the GSD advisories  associated with a particular ecosystem?

### DISCUSSION THREAD

**Discussion by jasnow** [cloudsecurityalliance/gsd-tools#170] [Q&A]
Question: How does the GSD database team identify all the GSD advisories 
associated with a particular ecosystem, such as "ruby" languagie?
Probably looking for a one or more database fields or
one or more keywords/key-phases.

**Comment by kurtseifried** [comment]
Simple answer:

OSV data has that
CVE sort of has it, products, so we'll need product->ecosystem
also source URLs -> ecosystem are typically good indicator 
and then good old "manual" setting of it (although in today's world that probably means fine tunning a ChatGPT model).

**Comment by joshbuker** [comment]
The easiest way to pull the ecosystem for an ID (e.g., `RubyGems`) would be from the OSV data.

For example,
```ruby
require 'json'
require 'net/http'
require 'uri'

api_response = Net::HTTP.get(URI.parse('https://api.gsd.id/GSD-2023-1002423'))
json_data = JSON.parse(api_response)
ecosystem = json_data['gsd']['osvSchema']['affected'][0]['package']['ecosystem']

puts ecosystem
puts (ecosystem == 'RubyGems').to_s
```

```text
Linux
false
```

See the [OSV Schema](https://ossf.github.io/osv-schema/#affectedpackage-field) for currently supported Ecosystems.

**Reply by joshbuker** [reply]
I just saw https://github.com/G-Rath/osv-detector/issues/184, and noticed that you're right - OSV doesn't support identifying issues with `Ruby` itself for example.

We also support `cve5Schema`, which might be better able to capture that data. The intention being anything that OSV lacks can be represented via CVE5 data instead.

**Reply by jasnow** [reply]
Also jruby, mruby, rbx, etc.

**Reply by joshbuker** [reply]
@jasnow This might be worth mentioning in the discussion on what API endpoints GSD should support (searching by ecosystem/gem/rubies): https://github.com/cloudsecurityalliance/gsd-tools/discussions/167

**Comment by jasnow** [comment]
Created a osv-schema issue earlier this morning for add another type of Ecosystem: https://github.com/ossf/osv-schema/issues/123
